### PR TITLE
gh-107442: Document all valid types for ctypes `_as_parameter_`

### PR DESCRIPTION
**Pull request by tomasr8**
* Issue: gh-107442

----
:books: Documentation preview :books:: https://cpython-previews--107443.org.readthedocs.build/

### CODE REVIEW

**Review by kamilturek — APPROVED** [review]

File: Doc/library/ctypes.rst [hunk 1]
```diff
@@ -400,9 +400,10 @@ Calling functions with your own custom data types
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 You can also customize :mod:`ctypes` argument conversion to allow instances of
-your own classes be used as function arguments.  :mod:`ctypes` looks for an
-:attr:`_as_parameter_` attribute and uses this as the function argument.  Of
-course, it must be one of integer, string, or bytes::
+your own classes be used as function arguments. :mod:`ctypes` looks for an
+:attr:`_as_parameter_` attribute and uses this as the function argument. The
+attribute must be an integer, string, bytes, a :mod:`ctypes` instance, or an
+object with an :attr:`_as_parameter_` attribute::
```
Comment (erlend-aasland) [inline Doc/library/ctypes.rst]: BTW, we try to add links only for the first reference in a paragraph. Also, it is not needed to link to the ctypes page from within the ctypes page, _and_ AFAICS, the *_as_parameter* should never be linked to :)

```suggestion
your own classes be used as function arguments. :mod:`!ctypes` looks for an
:attr:`!_as_parameter_` attribute and uses this as the function argument. The
attribute must be an integer, string, bytes, a :mod:`!ctypes` instance, or an
object with an :attr:`!_as_parameter_` attribute::
```

**Comment by erlend-aasland** [comment]
BTW, I created #107672 for the other annotation warnings in ctypes.rst.

File: Misc/NEWS.d/next/Documentation/2023-07-29-20-17-47.gh-issue-107442.L72LHv.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Document all valid types for the ``_as_parameter_`` attribute in
```
Comment (erlend-aasland) [inline Misc/NEWS.d/next/Documentation/2023-07-29-20-17-47.gh-issue-107442.L72LHv.rst]: Use correct markup for attributes:

```suggestion
Document all valid types for the :attr:`!_as_parameter_` attribute in
```

**Comment by erlend-aasland** [comment]
Come to think of it, we don't usually add NEWS entries for doc updates; could you please remove the NEWS entry? (Sorry for not mentioning this earlier!)

**Review by erlend-aasland — APPROVED** [review]

**Comment by erlend-aasland** [comment]
@tomasr8, do you know how to use the cherry-picker for the 3.11 backport?

**Comment by tomasr8** [comment]
> @tomasr8, do you know how to use the cherry-picker for the 3.11 backport?

I have never used the tool before, but I'll give it a try :)

**Comment by erlend-aasland** [comment]
There's some text about it in the devguide ;)

**Pull request merged**